torch/package/package_exporter.py

```diff
@@ -585,24 +586,42 @@ def save_pickle(
             is_pickle=True,
         )
 
+        def _check_mocked_error(module: str, field: str):
+            for pattern, pattern_info in self.patterns.items():
+                if pattern_info.action == _ModuleProviderAction.MOCK:
+                    if pattern.matches(module):
+                        raise NotImplementedError(
+                            f"Object '{field}' from module {module} was mocked out during packaging "
+                            f"but is being used in resource - {resource} in package {package}"
+                            "If this error is happening during 'save_pickle', please ensure that your "
+                            "pickled object doesn't contain any mocked objects."
+                        )
+
         if dependencies:
             all_dependencies = []
+            module = None
+            field = None
             for opcode, arg, pos in pickletools.genops(data_value):
-                if opcode.name == "GLOBAL":  # a global reference
+                if sys.version_info >= (3, 4):
```

Comment:
how do we know this is the only way that the arguments to STACK_GLOBAL get set up?

Reply:
this is a versioning thing, we enforce pickle 4 from python 3.4 onwards.